tests/data/cases/line_ranges_diff_edge_case.py

```diff
@@ -0,0 +1,28 @@
+# flags: --line-ranges=10-11
+# NOTE: If you need to modify this file, pay special attention to the --line-ranges=
+# flag above as it's formatting specifically these lines.
+
+# Reproducible example for https://github.com/psf/black/issues/4033.
+# This can be fixed in the future if we use a better diffing algorithm, or make Black
+# perform formatting in a single pass.
+
+print ( "format me" )
+print ( "format me" )
+print ( "format me" )
+print ( "format me" )
+print ( "format me" )
+
+# output
+# flags: --line-ranges=10-11
+# NOTE: If you need to modify this file, pay special attention to the --line-ranges=
+# flag above as it's formatting specifically these lines.
+
+# Reproducible example for https://github.com/psf/black/issues/4033.
+# This can be fixed in the future if we use a better diffing algorithm, or make Black
+# perform formatting in a single pass.
+
+print ( "format me" )
+print("format me")
+print("format me")
+print("format me")
```

Comment:
Why are these lines formatted? Shouldn't it only format the second and third prints?

Reply:
This is caused exactly by the diff algorithm used in `adjusted_lines`. In the second formatting pass, the adjusted lines becomes 12-13 where it should stayed as 10-11.

This PR doesn't fix this underlying issue (I'll file a separate issue if this PR is merged). Until it's really fixed, I'm proposing to skip the stability check (affected by the same issue) to avoid the crash.

The edge case is very rare in real code as it requires many same unformatted lines.

Reply:
I see, thanks. Maybe in that case we should put a warning in the docs for now about this bug.

Reply:
Agree that this doesn't feel very likely in real code, though @tartley apparently did run into this.

Reply:
Updated the doc.

@tartley can correct me, but #4033 appears to be a "toy" example for just trying out the new `--line-ranges` feature. #4052 can also serves the purpose of collecting real world example in case it happens.

What do you think?

Reply:
Agreed that I only encountered the problem using toy example code. By bad luck the toy example was the first thing I tried `--line-ranges` out on, while trying to integrate the use of it into my editor (e.g. so Vim would call Black on the selected range of lines). Although only a toy example it did genuinely confuse the heck out of me, and convinced me at the time that `--line-ranges` wasn't working for me at all.